pages/index/IndexHandler.inc.php

```diff
@@ -102,10 +102,9 @@ function _displayPressIndexPage($press, &$templateMgr) {
 			$enableAnnouncementsHomepage = $press->getSetting('enableAnnouncementsHomepage');
 			if ($enableAnnouncementsHomepage) {
 				$numAnnouncementsHomepage = $press->getSetting('numAnnouncementsHomepage');
-				$templateMgr->assign('enableAnnouncementsHomepage', true);
 				$announcementDao = DAORegistry::getDAO('AnnouncementDAO');
 				$announcements =& $announcementDao->getAnnouncementsNotExpiredByAssocId(ASSOC_TYPE_PRESS, $press->getId());
-				$templateMgr->assign_by_ref('announcements', $announcements);
+				$templateMgr->assign_by_ref('announcements', $announcements->toArray());
```

Comment:
Change `assign_by_ref` to `assign` -- references are rarely useful anymore since we've moved to modern PHP.